### ISSUE TITLE
rename `someTypePart` callback parameters and types aliases

### ISSUE THREAD

**Issue by RebeccaStevens** [JoshuaKGoldberg/ts-api-utils#79]
## Overview

This function takes two callback functions, one referred to as a "predicate" and the other one simply as "callback". But these functions are both predicate callback functions. So the names are a bit confusing.

I think we need to rename these functions.

My thinking is to refer to the first as the "type predicate" and the second as either the "result predicate" or the "test predicate". My reasoning being that the first one is used to test `type` before the second is then used get the actual return value.

We might be able to come up with better names though.

**Comment by JoshuaKGoldberg** [comment]
Aha now that you mention it, I remember getting confused by this! +1.

My internal naming for `predicate` was to think of it as `isUnionOrIntersectionType`. 

...but, why have the `predicate` in the first place? Internally `predicate` is only ever `isUnionOrIntersectionType` Per https://sourcegraph.com/search?q=context:global+someTypePart+lang:TypeScript+-file:.*node_modules.*+-file:.*%5C.d%5C.ts&patternType=standard&sm=1&groupBy=repo it's never used outside of wotan, and set to `isUnionType` there. Which could be re-implemented with only a couple of extra lines of code to not use `someTypePart`. 

Suggestion: let's remove `someTypePart`?

**Comment by RebeccaStevens** [comment]
> Suggestion: let's remove `someTypePart`?

Yeah, we could do that. Though this function is publicly exposed to users.
